cpp/src/gandiva/tests/filter_test.cc

```diff
@@ -414,6 +414,39 @@ TEST_F(TestFilter, TestLike) {
 
   // Validate results
   EXPECT_ARROW_ARRAY_EQUALS(exp, selection_vector->ToArray());
+
+  auto literal_escape_pattern =
+      TreeExprBuilder::MakeStringLiteral("%tu^_security^_freeze%");
+  auto escape_char = TreeExprBuilder::MakeStringLiteral("^");
```

Comment:
Can we use a PCRE special character that is invalid without suitable escap such as `[` instead?

Reply:
The escape char used here ('^') is one of the pcre spacial characters.